pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/MetadataStoreConfig.java

```diff
@@ -81,6 +84,39 @@ public class MetadataStoreConfig {
     @Builder.Default
     private final String metadataStoreName = "";
 
+    public static MetadataStoreConfigBuilder builder() {
+        return new CustomMetadataStoreConfigBuilder();
+    }
+
+    public static class CustomMetadataStoreConfigBuilder extends MetadataStoreConfigBuilder {
+
+        private String callerSetMetadataStoreName;
+
+        @Override
+        public MetadataStoreConfigBuilder metadataStoreName(String metadataStoreName) {
+            this.callerSetMetadataStoreName = metadataStoreName;
+            return super.metadataStoreName(metadataStoreName);
+        }
+
+        @Override
+        public MetadataStoreConfig build() {
+            if (Strings.isEmpty(this.callerSetMetadataStoreName)) {
+                Exception e = new Exception("stack trace");
```

Comment:
I think you can throw an exception, so like:
```
throw new IllegalArgumentException("metadataStoreName is required");
```

Reply:
I agree with you. the problem is there is a lot of ut didn't set name. If they don't set name but they close the object the flaky test won't occur.

Reply:
So can we just need to close the metadatastore to fix the flaky test?

Reply:
yes, other modification is for this case not happen again and supply some method to find the wrong usage.

the modification won't hurt production.

Reply:
> yes, other modification is for this case not happen again and supply some method to find the wrong usage.

Makes sense! But this is rare.

Reply:
yes i agree with you. maybe we can add one comment in this unit test point to the current modification to catch up new wrong usage. and the branch will be retain for those who is block by this problem.

I'll remove the current metadataStore name generate code for this patch to merge.